frontend/src/libs/components/input/input.tsx

```diff
@@ -5,36 +5,62 @@ import {
   type FieldValues,
 } from 'react-hook-form';
 
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
 import { useFormController } from '~/libs/hooks/hooks.js';
 
+import styles from './styles.module.scss';
+
 type Properties<T extends FieldValues> = {
   control: Control<T, null>;
   errors: FieldErrors<T>;
-  label: string;
+  label?: string;
   name: FieldPath<T>;
   placeholder?: string;
   type?: 'text' | 'email' | 'password';
+  isDisabled?: boolean;
 };
 
 const Input = <T extends FieldValues>({
   control,
   errors,
-  label,
+  label = '',
   name,
   placeholder = '',
   type = 'text',
+  isDisabled,
 }: Properties<T>): JSX.Element => {
   const { field } = useFormController({ name, control });
 
   const error = errors[name]?.message;
   const hasError = Boolean(error);
+  const hasValue = Boolean(field.value);
+  const hasLabel = Boolean(label);
+  const inputStyles = [
+    styles.input,
+    hasValue && styles.active,
+    isDisabled && styles.disabled,
+    error && styles.error,
+  ];
 
   return (
-    <label>
-      <span>{label}</span>
-      <input {...field} type={type} placeholder={placeholder} />
-      {hasError && <span>{error as string}</span>}
-    </label>
+    <div className={styles.inputComponentWrapper}>
+      {hasLabel && <span className={styles.label}>{label}</span>}
+      <div className={styles.inputWrapper}>
+        <input
```

Comment:
I believe it still should be wrapped with `<label>` tag (just change one of the div's to label, but don't forget that dev can't be used as child of label)

Reply:
fixed. replaced outter div to label, and inner div to span